### ISSUE TITLE
Memory usage keeps increasing

### ISSUE THREAD

**Issue by Gadi-Manor** [moleculerjs/moleculer#427]
We have a service that It's memory utilization keeps raising even when it pretty much doesn't do anything. Eventually the usage is so high what makes the service to crash.
I created a minimized service that all it does is creates a `ServiceBroker` and activates another "suspicious" module. 
I created the service with the following parameters:
 `logLevel: 'debug',`
 ` logger: myLogger,`
`transit: {
              disableReconnect: true
            },`
`transporter: {
    type: 'MQTT',
    options: {
      username: 'admin',
      password: 'admin!',
      servers: [
        {
          host: 'localhost',
          port: 1883
        }
      ]
    }
  }`

After a few hours I could tell that the memory issue still occurs although nothing is done by the service and the only thing written to the logs are the `HEARTBEAT` messages.
Then, I removed the part that creates the `ServiceBroker` and run the service again. 
Now the memory usage is much more steady. 
![memory](https://user-images.githubusercontent.com/41996966/49570603-c3eac900-f93f-11e8-9370-af84dc63abf2.PNG)
Any thoughts on what can cause it?

**Comment by icebob** [comment]
Please check it with another transporter (e.g. NATS) and without `disableReconnect`. I think one of them will cause it.

**Comment by tinchoz49** [comment]
Another important thing: what are you using in `myLogger`?

**Comment by Gadi-Manor** [comment]
> Another important thing: what are you using in myLogger?

[That](https://github.com/winstonjs/winston)

> Please check it with another transporter (e.g. NATS) and without disableReconnect. I think one of them will cause it.

I'm currently running the service with `TCP` transporter* and I will update on the result. In the meanwhile I looked in `mqtt.js` and saw there is an open issue about memory leak so maybe that's the problem.

* I know it's not ready for production, it just faster for me to test it then `NATS`.

**Comment by Gadi-Manor** [comment]
I tried different transporters (`NATS` and `TCP`), I also removed the `logger` parameter so no logs are showing at all but in all cases I'm still getting a constant climb in memory usage. Any other thoughts you think I should try?

**Comment by Gadi-Manor** [comment]
Found it!
**TL:DR** - The memory leak wasn't in `Moleculer`.

The scenario:
As I said before, I created a very simple service containing `Moleculer` and another module and played with the configuration, e.g. used both of them or only one to see how it affects the memory utilization.
What I've found was when `Moleculer` was used there was a memory leak and when it wasn't used there wasn't, so It's gotta be `Moleculer` right?

Well, it doesn't. It turns out that this other module used [this](https://www.npmjs.com/package/trace-and-clarify-if-possible) while on `debug` mode (which by accident it was). I didn't go very deep to the code but from what I understand it keep trace of the application and since while I was using `Moleculer` there was much more stuff going on than when I wasn't the memory leak was more obvious.

Sorry for wasting your time :)

**Comment by tinchoz49** [comment]
Thanks for sharing what you found it! :+1:
